### PR TITLE
Catch navigator.clipboard errors

### PR DESCRIPTION
**Pull request by spacegaier**
## Proposed change

Follow-up based on this comment: https://github.com/home-assistant/frontend/pull/7900#discussion_r538236792

Cannot really test, since my dev instance is not using HTTPS => `navigator.clipboard` is undefined, so legacy coding is always used.

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [X] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [X] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/common/util/copy-clipboard.ts [hunk 1]
```diff
@@ -1,12 +1,17 @@
-export const copyToClipboard = (str) => {
+export const copyToClipboard = async (str) => {
```
Comment (bramkragten) [inline src/common/util/copy-clipboard.ts:1]: The functions that use this helper should not also await it, since it is async.

Comment (spacegaier) [reply src/common/util/copy-clipboard.ts:1]: Still trying to wrap my head around the async/await stuff, but could it be that it's a typo and you meant "no**w** also await"?

Comment (bramkragten) [reply src/common/util/copy-clipboard.ts:1]: yes

File: src/panels/config/info/system-health-card.ts [hunk 2]
```diff
@@ -197,7 +197,7 @@ class SystemHealthCard extends LitElement {
     });
   }
 
-  private _copyInfo(ev: CustomEvent<ActionDetail>): void {
+  private async _copyInfo(ev: CustomEvent<ActionDetail>) {
```
Comment (bramkragten) [inline src/panels/config/info/system-health-card.ts:200]:
```suggestion
  private async _copyInfo(ev: CustomEvent<ActionDetail>): Promise<void> {
```

File: src/panels/config/automation/ha-automation-editor.ts [hunk 3]
```diff
@@ -397,7 +397,7 @@ export class HaAutomationEditor extends KeyboardShortcutMixin(LitElement) {
 
   private async _copyYaml() {
     if (this._editor?.yaml) {
-      copyToClipboard(this._editor.yaml);
+      await copyToClipboard(this._editor.yaml);
```
Comment (bramkragten) [inline src/panels/config/automation/ha-automation-editor.ts:400]: If we don't do anything after, there is no need to await it, we should just await it when we are going to show a toast that it is done etc.

**Review by bramkragten — APPROVED** [review]

**Pull request merged**